src/utils/context.rs

```diff
@@ -21,8 +23,18 @@ impl<'a, 'r> Context<'a, 'r> {
         self.get::<ServiceCache>()
     }
 
-    pub fn host(&self) -> Option<&str> {
-        self.request.headers().get_one("Host")
+    pub fn host(&self) -> Option<String> {
+        self.request.headers().get_one("Host").map(|host| {
+            if host.starts_with("localhost:") {
+                String::from(host)
+            } else {
+                format!("{}://{}", scheme(), host)
```

Comment:
Not sure if I would include the scheme here, I wouls leave it as plain host and prepend it when building the absolute uri as even for localhost a scheme should be defined

Reply:
hmm I tried prepending it for localhost but firefox wouldn't recognise the string as a url